test/unittest/test_scheduler.hpp

```diff
@@ -0,0 +1,49 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TEST_SCHEDULER_HPP
+#define LIBCYPHAL_TEST_SCHEDULER_HPP
+
+#include <libcyphal/types.hpp>
+
+namespace libcyphal
+{
+
+struct TestScheduler final
```

Comment:
this doesn't test the scheduler. It's a test fixture. Please use the following testing terminology:

test == tests
fixture == generic term for anything used to setup code to be tested
dummy or fake == an fixture designed to provide isolated dependencies to code under test
mock == dummy or fake that is instrumented and which can have assertions made against its use as part of a test.

Give these I'd call this a Fake or DummyScheduler within the `scheduler_fixtures.hpp` file.

Reply:
I'll rename it `VirtualTimeScheduler`.